src/relations/peers.py

```diff
@@ -121,6 +159,8 @@ def _on_created(self, event: RelationCreatedEvent):
 
     def _on_changed(self, event: RelationChangedEvent):
         """If the current unit is a follower, write updated config and auth files to filesystem."""
+        self.unit_databag[ADDRESS_KEY] = self.charm.unit_pod_hostname
```

Comment:
might be good to set this in the `install` hook as well, otherwise it might only get initialised when receiving `on_changed` events.

Reply:
`on_changed` is pretty common, but I'll add it to the `on_joined` hook anyway :)